### PR TITLE
Fix allure multiremote

### LINKED ISSUE #3442: Allure throws an error for Multi-remote tests
If you choose your reporter as `allure` i.e `reporters: ['spec', 'allure' ]` for a Multi-remote test, the allure-reporter package throws an error. 

```
TypeError: Cannot destructure property browserNameof 'undefined' or 'null'. 
[MultiremoteBrowser #0-0] at <Jasmine> 
[MultiremoteBrowser #0-0] at emitOne (events.js:116:13) 
[MultiremoteBrowser #0-0] at AllureReporter.emit (events.js:211:7)
``` 

This above error happens for any simple Multi-remote test like 

```
capabilities: {
        chromeDriver: {
            capabilities: {
                browserName: config.BROWSER_NAME.CHROME
            }
        },
        firefoxDriver: {
            capabilities: {
                browserName: config.BROWSER_NAME.FIREFOX
            }
        }
}
```

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

fix #3442 

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

added 1 test, moved duplicated code to beforeEach block

### CODE REVIEW

File: packages/wdio-allure-reporter/src/index.js [hunk 1]
```diff
@@ -46,12 +46,16 @@ class AllureReporter extends WDIOReporter {
...
-        const paramValue = version ? `${targetName}-${version}` : targetName
-        currentTest.addParameter('argument', paramName, paramValue)
+        if (!this.isMultiremote) {
+            const {browserName, deviceName} = this.config.capabilities
+            const targetName = browserName || deviceName || test.cid
+            const version = this.config.capabilities.version || this.config.capabilities.platformVersion || ''
+            const paramName = deviceName ? 'device' : 'browser'
+            const paramValue = version ? `${targetName}-${version}` : targetName
+            currentTest.addParameter('argument', paramName, paramValue)
+        } else {
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/src/index.js]: Can we have a test for when multiremote is true?

Comment (mgrybyk) [reply packages/wdio-allure-reporter/src/index.js]: @christian-bromann we have it here https://github.com/webdriverio/webdriverio/pull/3444/files#diff-7f34dcabc90a2eacc87fda48010c7a70
line `expect(addParameter).toHaveBeenCalledWith('argument', 'isMultiremote', 'true')`

File: packages/wdio-allure-reporter/tests/allureFeatures.test.js [hunk 2]
```diff
@@ -275,6 +235,47 @@ describe('reporter runtime implementation', () => {
...
+            })
+
+            reporter.allure = {
+                startCase: mock,
+                getCurrentSuite: mock,
+                getCurrentTest: mock,
+            }
+        })
+
+        it('should correctly add argument for selenium', () => {
```
Comment (mgrybyk) [inline packages/wdio-allure-reporter/tests/allureFeatures.test.js]: existing test, just moved out same code to beforeEach

File: packages/wdio-allure-reporter/tests/allureFeatures.test.js [hunk 3]
```diff
@@ -275,6 +235,47 @@ describe('reporter runtime implementation', () => {
...
+            reporter.onRunnerStart({config: {capabilities: {browserName: 'firefox', version: '1.2.3'}}})
+            reporter.onTestStart({cid: '0-0', title: 'SomeTest'})
+            expect(addParameter).toHaveBeenCalledTimes(1)
+            expect(addParameter).toHaveBeenCalledWith('argument', 'browser', 'firefox-1.2.3')
+        })
+
+        it('should correctly add argument for appium', () => {
```
Comment (mgrybyk) [inline packages/wdio-allure-reporter/tests/allureFeatures.test.js]: existing test, just moved out same code to beforeEach

File: packages/wdio-allure-reporter/tests/allureFeatures.test.js [hunk 4]
```diff
@@ -275,6 +235,47 @@ describe('reporter runtime implementation', () => {
...
+            reporter.onRunnerStart({config: {capabilities: {deviceName: 'Android Emulator', platformVersion: '8.0'}}})
+            reporter.onTestStart({cid: '0-0', title: 'SomeTest'})
+            expect(addParameter).toHaveBeenCalledTimes(1)
+            expect(addParameter).toHaveBeenCalledWith('argument', 'device', 'Android Emulator-8.0')
+        })
+
+        it('should correctly add argument for multiremote', () => {
```
Comment (mgrybyk) [inline packages/wdio-allure-reporter/tests/allureFeatures.test.js]: new test for multiremote

**Pull request merged**